cloudinit/config/schema.py

```diff
@@ -775,7 +821,7 @@ def validate_cloudconfig_file(
                     cloudconfig,
                     content,
                     marks,
-                    schema_errors=e.schema_errors,
+                    schema_errors=errors + e.schema_errors,
```

Comment:
Just based on typing, `e.schema_errors` can return `None`, so it's possible we could get a `TypeError` here adding a list to None. However, I haven't fully grokked all error paths, so if that's not a realistic worry, feel free to ignore.